app/code/Magento/InventoryCatalog/Model/ResourceModel/SetDataToLegacyStockItem.php

```diff
@@ -46,8 +46,11 @@ public function __construct(
      */
     public function execute(string $sku, float $quantity, int $status)
     {
-        $productId = $this->getProductIdsBySkus->execute([$sku])[$sku];
-
+        $productIds = $this->getProductIdsBySkus->execute([$sku]);
+        if (count($productIds) === 0) {
```

Comment:
W can not do it
If we skip this check we will have situation like new MSI data is updated but legacy data is not updated

Reply:
Looks like we need to add one more validator
https://github.com/magento-engcom/msi/issues/270